core/user_settings.py

```diff
@@ -59,6 +44,40 @@ def get_list_from_csv(
 
     return mapping
 
+def write_csv_defaults(path: Path, headers: tuple[str, str], default: dict[str, str]=None) -> None:
+    if not path.is_file() and default is not None:
+        with open(path, "w", encoding="utf-8", newline="") as file:
+            writer = csv.writer(file)
+            writer.writerow(headers)
+            for key, value in default.items():
+                writer.writerow([key] if key == value else [value, key])
+
+def track_csv_list(filename: str, headers: tuple[str, str], default: dict[str, str]=None) -> DecoratorT:
+    assert filename.endswith(".csv")
+    path = SETTINGS_DIR / filename
+    write_csv_defaults(path, headers, default)
+
+    def decorator(fn: CallbackT) -> CallbackT:
+        @resource.watch(str(path))
+        def on_update(f):
```

Comment:
Should we use [`wraps`](https://docs.python.org/3/library/functools.html#functools.wraps) here?

Reply:
no, we're not actually modifying the function.

however we might want to add `return fn` at the end of the decorator.

Reply:
Ah right yeah. Either that or update the return type of the decorator